MyToDo/SearchBar/SearchBarView.swift

```diff
@@ -0,0 +1,66 @@
+//
+//  SearchBarView.swift
+//  MyToDo
+//
+//  Created by Ivan Ruiz Monjo on 20/2/21.
+//
+
+import SwiftUI
+
+struct SearchBarView: View {
+
+    @State private var search: String = ""
+    @State private var isSearching: Bool = false
+    var searchBlock: (String) -> ()
+    var body: some View {
+        ZStack {
+            Color(#colorLiteral(red: 1, green: 1, blue: 1, alpha: 1))
+            HStack {
+                HStack {
+                    Image(systemName: "magnifyingglass")
+                        .imageScale(.small)
+                    TextField("", text: $search)
+                        .onChange(of: search) { searchText in
+                            searchBlock(searchText)
```

Comment:
I'm not entirely sure how I'd tackle this in SwiftUI but currently you search for every typed character. It's common to debounce search input by waiting for a user to briefly stop typing (often 300ms) before performing a search. Another common thing to do is removing duplicates.

The basic Combine code would look a bit like this:

```
cancellable = somePublisher
  .debounce(for: .seconds(0.3), scheduler: DispatchQueue.main)
  .removeDuplicates()
  .sink(receiveCompletion: { _ in }, receiveValue: { _ in }
```

I'm sure it should be possible to debounce the call to `searchBlock` in a similar way. I'm just not sure how

Reply:
Let me show you my solution tomorrow 👍